apis/extension/node_reservation.go

```diff
@@ -0,0 +1,47 @@
+/*
+Copyright 2022 The Koordinator Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package extension
+
+import corev1 "k8s.io/api/core/v1"
+
+const (
+	ReservedByNode = NodeDomainPrefix + "/node-reservation"
+)
+
+// KoordReserved resource reserved by node.annotation,
+// If node.annotation declares the resources to be reserved, like this:
+//  annotations:
+//    node.koordinator.sh/node-reservation: >-
+//	    {"reservedCPUs":"0-5","qosEffected":["LSE"]}
+
+// if pod.qos==LSE
+//   In the filter phase it needs to satisfy: node.alloc - node.req - reserved(6c) > pod.req
+//   the cores 0-5 are not used in the reserve phase.
+// if pod.qos==LSR
+//   In the filter phase it needs to satisfy: node.alloc - node.req > pod.req
+//   the cores 0-5 are not used in the reserve phase.
+// if pod.qos!=LSE && pod.qos!=LSR
+//	 In the filter phase it needs to satisfy: node.alloc - node.req > pod.req
+type KoordReserved struct {
```

Comment:
The name `KoordReserved` is too generic, there should be a more expressive name, consider `NodeReservation` ?

Reply:
> There is another special place that needs to be modified. The code related to elasticQuota will perceive Node.Status.Allocatable as the total amount that can be allocated by Quota. Here we also need to modify it.

done